frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1978,14 +1958,17 @@
                     "method_description": vm.method_description,
                     "project_url": vm.project_url,
                     "publication_url": vm.publication_url,
-                    "submission_metadata": JSON.stringify(vm.submissionMetaAttributes[vm.submissionMetaAttributeIndex].attributes)
+                    "submission_metadata": vm.submission_metadata
                 };
                 parameters.callback = {
                     onSuccess: function(response) {
                         var status = response.status;
                         if (status === 200) {
                             $mdDialog.hide();
                             $rootScope.notify("success", "The data is successfully updated!");
+                            if(vm.submission_metadata != null) {
+                                vm.submissionMetaData.submission_metadata = JSON.parse(JSON.stringify(vm.submission_metadata));
```

Comment:
same question here. Why converting to str and then json back

Reply:
@Ram81, I want to create a clone of the json object which are independent of each other. This is to create a backup of the json object stored in database. Directly assigning a = b in case of json makes a link between a and b. If a is changed then b also changes.